src/tunacode/tools/bash.py

```
@@ -171,39 +179,42 @@ async def _run_bash(
-    except Exception as exc:  # noqa: BLE001
-        raise ToolExecutionError(tool_name="bash", message=str(exc), original_error=exc) from exc
+def build_bash_tool(*, max_command_output: int) -> AgentTool:
+    async def _execute_bash(
+        tool_call_id: str,
+        args: JsonObject,
+        signal: asyncio.Event | None,
+        on_update: AgentToolUpdateCallback,
+    ) -> AgentToolResult:
+        _ = (tool_call_id, on_update)
+        if signal is not None and signal.is_set():
+            raise UserAbortError("Tool execution aborted: bash")
```

Ruff (0.15.9)

[warning] Avoid specifying long messages outside the exception class

(TRY003)